MinimalPluginStandard/Sniffs/DirectDBSniff.php

```diff
@@ -178,30 +186,30 @@ protected function mark_sanitized_var( $stackPtr ) {
 		// Find the closure or function scope of the variable.
 		$context = $this->get_context( $stackPtr );
 
-		$var = $this->get_complex_variable( $stackPtr );
+		$var = $this->get_variable_as_string( $stackPtr );
+		$var = ltrim( $var, '$' );
```

Comment:
Why trim the `$`? At first I thought this was always being called right after `get_variable_as_string`, so it seemed like it should be included in that method, but then I saw that there are some places where you actually want to keep the `$` (like if it's printed in a log message). It seems like it would be good to keep it consistently one way or the other if possible.

Reply:
Good question. I was being lazy because variables regexed out of interpolated strings might look like `$foo` or `{$foo}` or `${foo}`. Trimming `${}` flowed from that, but you're right that it's confusing.

Fixed in 19715ff.